setup.py

```diff
@@ -1,109 +1,125 @@
 #!/usr/bin/python
-from setuptools import setup, Extension
 from Cython.Build import cythonize
+from setuptools import setup, Extension
 import os
-import sys
-import subprocess
+import pkgconfig
 import re
+import subprocess
+import sys
 
-hidapi_topdir = os.path.join("hidapi")
-hidapi_include = os.path.join(hidapi_topdir, "hidapi")
-system_hidapi = 0
-libs = []
-src = ["hid.pyx"]
+tld = os.path.abspath(os.path.dirname(__file__))
+embedded_hidapi_topdir = os.path.join(tld, "hidapi")
+embedded_hidapi_include = os.path.join(embedded_hidapi_topdir, "hidapi")
 
 
 def hidapi_src(platform):
-    return os.path.join(hidapi_topdir, platform, "hid.c")
+    return os.path.join(embedded_hidapi_topdir, platform, "hid.c")
 
 
-if "--with-system-hidapi" in sys.argv:
-    sys.argv.remove("--with-system-hidapi")
-    system_hidapi = 1
-    hidapi_include = "/usr/include/hidapi"
-
-if sys.platform.startswith("linux"):
-    modules = []
-    if "--without-libusb" in sys.argv:
-        sys.argv.remove("--without-libusb")
-        hidraw_module = "hid"
-    else:
-        hidraw_module = "hidraw"
-        libs = ["usb-1.0", "udev", "rt"]
-        if system_hidapi == 1:
-            libs.append("hidapi-libusb")
+def hid_from_embedded_hidapi():
+    # TODO: what about MinGW/msys?
+    if sys.platform.startswith("win") or sys.platform.startswith("cygwin"):
+        modules = [
+            Extension(
+                "hid",
+                sources=["hid.pyx", hidapi_src("windows")],
+                include_dirs=[embedded_hidapi_include],
+                libraries=["setupapi"],
+            )
+        ]
+
+    elif sys.platform.startswith("darwin"):
+        macos_sdk_path = (
+            subprocess.check_output(["xcrun", "--show-sdk-path"]).decode().strip()
+        )
+        modules = [
+            Extension(
+                "hid",
+                sources=["hid.pyx", hidapi_src("mac")],
+                include_dirs=[embedded_hidapi_include],
+                # TODO: -Wno-unreachable-code: https://github.com/cython/cython/issues/3172
+                extra_compile_args=['-isysroot', macos_sdk_path, '-Wno-unreachable-code'],
+                # TODO: remove '-framework AppKit' after switching to 0.14.1 or newer
+                extra_link_args=['-framework', 'IOKit', '-framework', 'CoreFoundation', '-framework', 'AppKit']
```

Comment:
Switched from using environment variables to `extra_compile_args`/`extra_link_args`.
Couldn't find the exact reason why env variables should used, except the fact that it was introduced 12 years ago during the initial commit.